pkg/sidecar/sidecar_test.go

```diff
@@ -21,9 +26,284 @@ import (
+		// is the raw OS exit code. These tests don't currently define
+		// a way to test for signal exits.
+		expectExitStatus int
+		expectSignalExit syscall.Signal
+		// Is this command expected to create a file? If so, the path
+		// to check for the file's existence.
+		expectFileExists string
+	}{
+		{
+			// Check that the command runs and exits, and
+			// we can observe a side effect to prove it ran
+			cmd:              "touch",
+			cmdArgs:          touchTestFile,
+			expectTerminated: true,
+			expectExitStatus: 0,
+			expectFileExists: touchTestFile,
+		},
+		{
+			// check nonzero exit code handling
+			cmd:              "false",
+			expectTerminated: true,
+			expectExitStatus: 1,
+		},
+		{
+			// run a sleep and wait for it to finish, ensuring that
+			// we actually wait for the exit of a process that takes
+			// a moment to run.
+			cmd:              "sleep",
+			cmdArgs:          "0.1",
+			expectTerminated: true,
+		},
+		{
+			// run a sleep and inspect state before it finishes
+			cmd:              "sleep",
+			cmdArgs:          "1",
+			expectTerminated: false,
+			timeout:          400 * time.Millisecond,
+		},
+		{
+			// signal exit handling - if a process exits with a signal
+			// we handle it gracefully
+			cmd:     "sh",
+			cmdArgs: "-c \"kill -TERM $$\"",
+			// Suppress the "signal: hangup" from the shell
+			stdout:           nil,
+			stderr:           nil,
+			expectTerminated: true,
+			expectSignalExit: syscall.SIGTERM,
+		},
+		{
+			// show that argument splitting is confusing - it will not handle single-quoted
+			// arguments how you might expect. This will fail with an unterminated string
+			// error (exit code 2) because the argument gets split into the vector
+			// ["-c", "'kill", "-TERM", "$$'"]. This shows that we
+			// should really be using an argument vector to accept
+			// commands, not a string.
+			cmd:     "sh",
+			cmdArgs: "-c 'kill -TERM $$'",
+			// Suppress the "TERM: 1: Syntax error: Unterminated
+			// quoted string" error message from the shell
+			stdout:           nil,
+			stderr:           nil,
+			expectTerminated: true,
+			expectExitStatus: 2,
+		},
+	}
+	for _, tc := range testcases {
+		tc := tc
+		t.Run(tc.cmd, func(t *testing.T) {
+			if runtime.GOOS == "windows" && tc.expectSignalExit != 0 {
+				t.Skip("Signal handling is not supported on Windows")
+			}
+
+			// Set up the harness for this sidecar command run
+			s := newSidecarTest(t)
+			s.NewConfig(t)
+			config := s.Config()
+			config.Cmd = tc.cmd
+			config.CmdArgs = tc.cmdArgs
+
+			s.NewSidecar(t)
+			// deliberately does not defer s.Close() since we might be abandoning
+			// the sidecar when it is still running a command, and don't want panics
+			// writing to unclosed channels.
+			sidecar := s.Sidecar()
+
+			if tc.stdin != nil {
+				sidecar.stdin = tc.stdin
+			}
+			if tc.stdout != nil {
+				sidecar.stdout = tc.stdout
+			}
+			if tc.stderr != nil {
+				sidecar.stderr = tc.stderr
+			}
```

Comment:
I dont see any of the test cases setting these to a non nil value.

Reply:
It's shared test harness code; trying to be defensive.

Reply:
Thanks for clarifying. Would you mind removing? It makes it hard for me to understand the code if there is stuff that isn't being used.

Reply:
I fixed and rewrote this per https://github.com/spiffe/spiffe-helper/pull/258#discussion_r1997748717 and the new commits pushed.